app/src/main/java/com/github/polypoly/app/ui/menu/lobby/GameLobbyActivity.kt

```diff
@@ -70,162 +83,623 @@ class GameLobbyActivity : ComponentActivity() {
         val gameLobby = gameLobbyWaitingModel.getGameLobby().observeAsState().value
         val readyForStart = gameLobbyWaitingModel.getReadyForStart().observeAsState().value
         val dataLoading = gameLobbyWaitingModel.getIsLoading().observeAsState().value
+        val context = LocalContext.current
 
-        if (gameLobby != null && readyForStart != null) {
+        if (gameLobby != null && readyForStart != null){
+
+            remoteDB.addOnChangeListener<GameLobby>(gameLobby.code, "started_game_listener") {
+                if (it.started && it.admin.id != currentUser.id) {
+                    navigateToGame(context, gameLobby)
+                }
+            }
```

Comment:
This king of logic should be put in the `gameLobbyWaitingModel` instead of in the UI part. You could then expose a LiveData in the form of a boolean such as isGameReadyforStart that you could observe here (= in the UI) to know if you need to launch `navigateToGame`

Reply:
this works for now. I will add a todo to refactor it and make clean as you suggest!